lib/cache-utils.js

```diff
@@ -55,25 +55,26 @@ function wellKnownCachePath() {
 
 export async function getCacheKey() {
   const platform = process.env.RUNNER_OS;
-  const fileHash = await glob.hashFiles(await getScarbManifestPath());
+  const fileHash = await glob.hashFiles(await getScarbLockfilePath());
 
   if (!fileHash) {
     throw new Error(
-      "failed to cache dependencies: unable to hash Scarb.toml file",
+      "failed to cache dependencies: unable to hash Scarb.lock file",
     );
   }
 
   return `scarb-cache-${platform}-${fileHash}`.toLowerCase();
 }
 
-async function getScarbManifestPath() {
-  const { stdout, exitCode } = await exec.getExecOutput("scarb manifest-path");
+async function getScarbLockfilePath() {
+  const globber = await glob.create("**/Scarb.lock");
```

Comment:
Maybe we shouldn't look for `Scarb.lock` recursively and just restrain to the root directory?
I see that you pick the shortest one in terms of length (not sure what `lockfiles` is, is it a list of strings? 🤔), is it the lockfile with the shortest path?

Reply:
We should look recursively and use the first we find in parent dirs - to support run in a package from a workspace.

Reply:
Discussed this with @maciektr offline, root package will be chosen at least for now